mlflow/store/model_registry/file_store.py

```diff
@@ -509,9 +512,11 @@ def create_model_version(
                  created in the backend.
         """
 
-        def next_version(registered_model):
-            if registered_model.latest_versions:
-                return max([mv.version for mv in registered_model.latest_versions]) + 1
+        def next_version(registered_model_name):
+            path = self._get_registered_model_path(registered_model_name)
+            model_versions = self._list_model_versions_under_path(path)
```

Comment:
Since there is no filter for a deletion-marked version from within `_list_model_versions_under_path()`, the behavior we're looking for is to auto-increment over historical registered versions, right? 

Instance 1:
- create model & register version 1
- create model & register version 2
- mark version 2 for deletion
- create model & register version 3

Instance 2:
- create model & register version 1
- create model & register version 2
- mark version 2 for deletion
- create model & register version 2

Just verifying that our intention is Instance 1 behavior, right?

Reply:
Could we add a version assertion to this test to have coverage for whichever behavior is intended? https://github.com/mlflow/mlflow/blob/38e1d481f427b94af8f471e1658b4971ca6391b5/tests/tracking/test_model_registry.py#L299-L301

Reply:
The intended behavior is:

- create model & register version 1
- create model & register version 2
- mark version 2 for deletion
- create model & register version 1